taxcalc/behavior.py

```diff
@@ -407,6 +480,179 @@ def _validate_parameter_values(self):
                 if zero_sub and zero_inc:
                     raise ValueError(msg.format(year + self.start_year))
 
+    # not sure about revisions name
+    def _validate_parameter_names_types(self, reform):
+        """
+        Check validity of parameter names and parameter types used
+        in the specified reform dictionary.
+        """
+        # pylint: disable=too-many-branches,too-many-nested-blocks
+        data_names = set(self._vals.keys())
+        for year in sorted(reform.keys()):
+            for name in reform[year]:
+
+                ########################
+                # Policy() specific code -- Behavior does not need CPI logic
+                if name.endswith('_cpi'):
+                    raise ValueError('Why does a Behavior end with CPI?')
+                    if isinstance(reform[year][name], bool):
+                        pname = name[:-4]  # root parameter name
+                        if pname not in data_names:
+                            msg = '{} {} unknown parameter name'
+                            self.behavior_errors += (
+                                'ERROR: ' + msg.format(year, name) + '\n'
+                            )
+                        else:
+                            # check if root parameter is cpi inflatable
+                            if not self._vals[pname]['cpi_inflatable']:
+                                msg = '{} {} parameter is not cpi inflatable'
+                                self.behavior_errors += (
+                                    'ERROR: ' + msg.format(year, pname) + '\n'
+                                )
+                    else:
+                        msg = '{} {} parameter is not true or false'
+                        self.behavior_errors += (
+                            'ERROR: ' + msg.format(year, name) + '\n'
+                        )
+                ########################
+
```

Comment:
This won't be used by the `Behavior` class, but again, I thought it would be interesting to show how much code from the `Policy` class could simply be copied over.